WordPress/Classes/ViewRelated/Reader/ReaderMenuViewModel.swift

```diff
@@ -0,0 +1,615 @@
+import Foundation
+import Gridicons
+
+
+/// Enum of the sections shown in the reader.
+///
+enum ReaderMenuSectionType: Int {
+    case Defaults
+    case Lists
+    case Tags
+}
+
+
+/// Enum of the types of row items shown in the reader.
+///
+enum ReaderMenuItemType: Int {
+    case Topic
+    case Search
+    case Recommended
+}
+
+
+/// Represents a section in the Reader's menu.
+///
+struct ReaderMenuSection {
+    let title: String
+    let type: ReaderMenuSectionType
+}
+
+
+/// Represents an row, or menu item in a reader menu section.
+///
+struct ReaderMenuItem
+{
+    let title: String
+    let type: ReaderMenuItemType
+    // A custom icon for the menu item.
+    var icon: UIImage?
+    // The corresponding topic for the item, if there is one (e.g. Search does not have a topic)
+    var topic: ReaderAbstractTopic?
+    // The order of the item if a custom order is used by the section.
+    var order: Int = 0
+
+
+    init(title: String, type: ReaderMenuItemType, icon: UIImage?, topic: ReaderAbstractTopic?) {
+        self.title = title
+        self.type = type
+        self.icon = icon
+        self.topic = topic
+    }
+
+
+    init(title: String, type: ReaderMenuItemType) {
+        self.title = title
+        self.type = type
+    }
+}
+
+
+/// Protocol allowing a reader menu view model to notify content changes.
+///
+protocol ReaderMenuViewModelDelegate
+{
+    /// Notifies the delegate that the menu is about to reload its content
+    ///
+    func menuWillReloadContent()
+
+
+    /// Notifies the delegate that the menu did reload its content.
+    ///
+    func menuDidReloadContent()
+
+
+    /// Notifies the delegate that the content of the specified section is about to change.
+    ///
+    /// - Parameters:
+    ///     - index: The index of the section.
+    ///
+    func menuSectionWillChangeContent(index: Int)
+
+
+    /// Notifies the delegate that the content of the specified section has changed.
+    ///
+    /// - Parameters:
+    ///     - index: The index of the section.
+    ///
+    func menuSectionDidChangeContent(index: Int)
+}
+
+
+/// Defines the preferred order of items in the default section.
+///
+enum ReaderDefaultMenuItemOrder: Int {
+    case Followed
+    case Discover
+    case Search
+    case Recommendations
+    case Likes
+    case Other
+}
+
+
+/// The view model used by the reader.
+///
+@objc class ReaderMenuViewModel: NSObject
+{
+    var defaultsFetchedResultsController: NSFetchedResultsController!
+    var listsFetchedResultsController: NSFetchedResultsController!
+    var tagsFetchedResultsController: NSFetchedResultsController!
```

Comment:
I considered it.  The original menu used a single results controller.  I went with separate controllers because I think it reduces some of the complexity in the default and, as you'll eventually see, the tags sections.